src/server/schedule/thread/pool.rs

```diff
@@ -94,19 +119,23 @@ impl Pool {
         Pool { _handles: handles, job_sender, parallelism: NonZero::new(threads).unwrap() }
     }
 
-    pub fn spawn<F>(&self, priority: ThreadPriority, f: F)
+    pub fn spawn<F>(&self, priority: ThreadPriority, f: F) -> TaskHandle
     where
         F: FnOnce() + Send + 'static,
     {
+        let handle = TaskHandle::new();
+        let handle_clone = handle.clone();
         let f = Box::new(move || {
             if cfg!(debug_assertions) {
                 priority.assert_is_used_on_current_thread();
             }
             f();
+            handle_clone.signal_finish();
         });
```

Comment:
```suggestion
        let handle = TaskHandle::new();
        let f = Box::new({
            let handle = handle.clone();
        
            move || {
                if cfg!(debug_assertions) {
                    priority.assert_is_used_on_current_thread();
                }
                f();
                handle_clone.signal_finish();
            }
        });
```

What do you think? Feel free to resolve.

Reply:
It's more clear to me as it was tbh